src/mococrw/ecies.h

```diff
@@ -0,0 +1,332 @@
+
+    /**
+     * @brief Sets the message authentication code used for ECIES. (Default: mococrw::HMAC with SHA512)
+     * @param mac The actual object used in ECIES
+     * @return builder instance
+     */
+    ECIESCtxBuilder &setMAC(std::shared_ptr<mococrw::MessageAuthenticationCode> mac);
+
+    /**
+     * @brief Sets the symmetric cipher mode. (Default: mococrw::SymmetricCipherMode::CBC)
+     * @param mode The mode used in ECIES
+     * @return builder instance
+     */
+    ECIESCtxBuilder &setSymmetricCipherMode(mococrw::SymmetricCipherMode mode);
+
+    /**
+     * @brief Sets the key size for the symmetric encryption. (Default: mococrw::SymmetricCipherKeySize::S_256)
+     * @param keySize The key size to be used
+     * @return builder instance
+     */
+    ECIESCtxBuilder &setSymmetricCipherKeySize(mococrw::SymmetricCipherKeySize keySize);
+
+    /**
+     * @brief Sets the padding for the symmetric encryption. (Default: mococrw::SymmetricCipherPadding::NO)
+     * @param padding The padding scheme to be used.
+     * @return builder instance
+     */
+    ECIESCtxBuilder &setSymmetircCipherPadding(mococrw::SymmetricCipherPadding padding);
+
+    /**
+     * @brief Sets the initialisation vector (IV) for the symmetric encryption. (Default: zeros)
+     * @param iv The initialisation vector. The length shall be the same as the key length
+     * @return builder instance
+     */
+    ECIESCtxBuilder &setSymmetricCipherIv(std::vector<uint8_t> iv);
+
+    /**
+     * @brief Sets the salt for the key derivation function (KDF). (Default: empty vector -> not used)
+     * @param kdfSalt The salt value used for key derivation.
+     * @return builder instance
+     */
+    ECIESCtxBuilder &setKDFSalt(std::vector<uint8_t> kdfSalt);
+
+    /**
+     * @brief Sets the salt for the message authentication code. (Default: empty vector -> not used).
+     * @param macSalt The salt value used for message authentication code.
+     * @return builder instance
+     */
+    ECIESCtxBuilder &setMACSalt(std::vector<uint8_t> macSalt);
+
+protected:
+    /**
+     * @brief constructor
+     *
+     * This class shall not be instantiated (except by subclasses)
+     */
+    ECIESCtxBuilder();
+
+private:
+    std::shared_ptr<mococrw::KeyDerivationFunction> kdf;
+    std::shared_ptr<mococrw::MessageAuthenticationCode> mac;
+    mococrw::SymmetricCipherMode mode;
+    mococrw::SymmetricCipherKeySize keySize;
+    mococrw::SymmetricCipherPadding padding;
+    std::vector<uint8_t> iv;
+    std::vector<uint8_t> kdfSalt;
+    std::vector<uint8_t> macSalt;
+
+};
+
+class ECIESEncryptionCtxBuilder : public ECIESCtxBuilder
+{
+public:
+    /**
+     * \brief Constructor for public keys
+     * \param bobsKey The public key of the receiver.
+     */
+    ECIESEncryptionCtxBuilder(const AsymmetricPublicKey& bobsKey);
+
+    /**
+     * @brief Constructor if the public key is inside a certificate
```

Comment:
```suggestion
     * @brief Constructor for certificates
```
That sounds odd... maybe just say certificates?